packages/bsky/src/hydration/label.ts

```diff
@@ -16,10 +17,36 @@ export type { Label } from '../lexicon/types/com/atproto/label/defs'
 
 export type SubjectLabels = {
   isTakendown: boolean
-  labels: Label[]
+  labels: HydrationMap<Label> // src + val -> label
 }
 
-export type Labels = HydrationMap<SubjectLabels>
+export class Labels extends HydrationMap<SubjectLabels> implements Merges {
+  static key(label: Label) {
+    return `${label.src}::${label.val}`
+  }
+  merge(map: Labels): this {
+    map.forEach((theirs, key) => {
+      if (!theirs) return
+      const mine = this.get(key)
+      if (mine) {
+        mine.isTakendown = mine.isTakendown || theirs.isTakendown
+        mine.labels = mine.labels.merge(theirs.labels)
+      } else {
+        this.set(key, theirs)
+      }
+    })
+    return this
+  }
+  getBySubject(sub: string): Label[] {
+    const it = this.get(sub)?.labels.values()
+    if (!it) return []
+    const labels: Label[] = []
```

Comment:
could be a nice fit for `mapDefined`?

Reply:
oh i guess it's an iterator. maybe not 🤔

Reply:
Ooh maybe `mapDefined()` should work on iterators...